tango_ros_common/tango_ros_native/src/tango_ros_conversions_helper.cpp

```diff
@@ -232,26 +234,62 @@ void toMeshMarker(const Tango3DR_GridIndex& grid_index,
   mesh_marker->id += grid_index[1];
   mesh_marker->id *= 37;
   mesh_marker->id += grid_index[2];
-  for (size_t j = 0; j < tango_mesh->num_faces; ++j) {
+  for (size_t j = 0; j < tango_mesh.num_faces; ++j) {
     // Add the 3 points of the triangle face and the corresponding colors.
     geometry_msgs::Point point;
-    toPoint(tango_mesh->vertices[tango_mesh->faces[j][0]], &point);
+    toPoint(tango_mesh.vertices[tango_mesh.faces[j][0]], &point);
     mesh_marker->points.push_back(point);
-    toPoint(tango_mesh->vertices[tango_mesh->faces[j][1]], &point);
+    toPoint(tango_mesh.vertices[tango_mesh.faces[j][1]], &point);
     mesh_marker->points.push_back(point);
-    toPoint(tango_mesh->vertices[tango_mesh->faces[j][2]], &point);
+    toPoint(tango_mesh.vertices[tango_mesh.faces[j][2]], &point);
     mesh_marker->points.push_back(point);
 
     std_msgs::ColorRGBA color;
-    toColor(tango_mesh->colors[tango_mesh->faces[j][0]], &color);
+    toColor(tango_mesh.colors[tango_mesh.faces[j][0]], &color);
     mesh_marker->colors.push_back(color);
-    toColor(tango_mesh->colors[tango_mesh->faces[j][1]], &color);
+    toColor(tango_mesh.colors[tango_mesh.faces[j][1]], &color);
     mesh_marker->colors.push_back(color);
-    toColor(tango_mesh->colors[tango_mesh->faces[j][2]], &color);
+    toColor(tango_mesh.colors[tango_mesh.faces[j][2]], &color);
     mesh_marker->colors.push_back(color);
   }
 }
 
+void toOccupancyGrid(const Tango3DR_ImageBuffer& image_grid,
+                     const Tango3DR_Vector2& origin,
+                     double time_offset,
+                     nav_msgs::OccupancyGrid* occupancy_grid) {
+  occupancy_grid->header.frame_id = toFrameId(TANGO_COORDINATE_FRAME_START_OF_SERVICE);
+  occupancy_grid->header.stamp.fromSec(image_grid.timestamp + time_offset);
+  occupancy_grid->info.map_load_time = occupancy_grid->header.stamp;
+  occupancy_grid->info.width = image_grid.width;
+  occupancy_grid->info.height = image_grid.height;
+  occupancy_grid->info.resolution = OCCUPANCY_GRID_RESOLUTION;
+  occupancy_grid->info.origin.position.x = origin[0];
+  // We have the position of the top left pixel, instead we want the position
+  // of the bottom left pixel.
+  occupancy_grid->info.origin.position.y = origin[1] - image_grid.height * OCCUPANCY_GRID_RESOLUTION;
+  for (size_t i = 0; i < image_grid.height; ++i) {
+    for (size_t j = 0; j < image_grid.width; ++j) {
+      // The image uses a coordinate system with (x: right, y: down), while
+      // the occupancy grid is using (x: right, y: up). The image is therefore
+      // flipped around the x axis.
+      int value = static_cast<int>(image_grid.data[j + (image_grid.height - i - 1) * image_grid.width]);
+      if (value == 1) {
+        // The cell is occupied.
+        occupancy_grid->data.push_back(100);
+      } else if (value == 255) {
+        // The cell is free.
+        occupancy_grid->data.push_back(0);
```

Comment:
Using meaningful names for constants instead of documentation for `0, -1, 100` would also work.
constexpr int FREE = 0 (same for UNKNOWN, OCCUPIED)